test/extended/networking/ovs.go

```diff
@@ -0,0 +1,314 @@
+			ip := strings.Split(ipPort, ":")[0]
+
+			newFlows := getFlowsForAllNodes(oc, nodes.Items)
+			for _, node := range nodes.Items {
+				if node.Name != deployNodeName {
+					Expect(reflect.DeepEqual(origFlows[node.Name], newFlows[node.Name])).To(BeTrue(), "Flows on non-deployed-to nodes should be unchanged")
+				}
+			}
+
+			var newDeployNodeFlows []string
+			var arpOut, ipOut, arpIn, ipInGeneric, ipInGlobal, ipInIsolated bool
+			for _, flow := range newFlows[deployNodeName] {
+				if strings.Contains(flow, ip) {
+					if strings.Contains(flow, "arp_spa="+ip) {
+						arpOut = true
+					} else if strings.Contains(flow, "arp_tpa="+ip) {
+						arpIn = true
+					} else if strings.Contains(flow, "nw_src="+ip) {
+						ipOut = true
+					} else if strings.Contains(flow, "nw_dst="+ip) {
+						if strings.Contains(flow, "reg0=0x") {
+							ipInIsolated = true
+						} else if strings.Contains(flow, "reg0=0") {
+							ipInGlobal = true
+						} else {
+							ipInGeneric = true
+						}
+					} else {
+						Fail("found unexpected OVS flow: " + flow)
+					}
+				} else {
+					newDeployNodeFlows = append(newDeployNodeFlows, flow)
+				}
+			}
+			Expect(arpOut).To(BeTrue(), "Should have an outgoing ARP rule")
+			Expect(arpIn).To(BeTrue(), "Should have an incoming ARP rule")
+			Expect(ipOut).To(BeTrue(), "Should have an outgoing IP rule")
+			if pluginIsolatesNamespaces() {
+				Expect(ipInGlobal && ipInIsolated).To(BeTrue(), "Should have global and isolated incoming IP rules")
+			} else {
+				Expect(ipInGeneric).To(BeTrue(), "Should have a generic incoming IP rule")
+			}
+			Expect(reflect.DeepEqual(origFlows[deployNodeName], newDeployNodeFlows)).To(BeTrue(), "Deploy node flows should be unchanged except for the new pod")
+
+			err := f1.Client.Pods(f1.Namespace.Name).Delete(podName, nil)
+			Expect(err).NotTo(HaveOccurred())
+
+			postDeleteFlows := getFlowsForNode(oc, deployNodeName)
+			Expect(reflect.DeepEqual(origFlows[deployNodeName], postDeleteFlows)).To(BeTrue(), "Flows after deleting pod should be same as before creating it")
+		})
+
+		It("should add and remove flows when nodes are added and removed", func() {
+			var err error
+			nodes := e2e.GetReadySchedulableNodesOrDie(f1.Client)
+			origFlows := getFlowsForAllNodes(oc, nodes.Items)
+
+			// The SDN/OVS code doesn't care that the node doesn't actually exist,
+			// but we try to pick an IP on our local subnet to avoid sending
+			// traffic into the real world.
+			highNodeIP := ""
+			for _, node := range nodes.Items {
+				if node.Status.Addresses[0].Address > highNodeIP {
+					highNodeIP = node.Status.Addresses[0].Address
+				}
+			}
+			Expect(highNodeIP).NotTo(Equal(""))
+			octets := strings.Split(highNodeIP, ".")
+			Expect(len(octets)).To(Equal(4))
+			lastOctet, err := strconv.Atoi(octets[3])
+			Expect(err).NotTo(HaveOccurred())
+			Expect(lastOctet).NotTo(Equal(255))
+			octets[3] = strconv.Itoa(lastOctet + 1)
+			newNodeIP := strings.Join(octets, ".")
```

Comment:
how about something like:

ip := net.ParseIP(highNodeIP)
Expect(ip).NotTo(BeNil())
Expect(ip.To4()).NotTo(BeNil())
ip[3] += 1
Expect(ip[3]).NotTo(Equal(255))
newNodeIP := ip.String()

instead of a bunch of text manipulation...